src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -346,6 +346,7 @@ public static byte[] byteArray() {
      *
      * @param size array size.
      * @return a new byte array of the given size.
+     * @throws NegativeArraySizeException if the size is negative.
```

Comment:
This PR seems ok to me now except for this addition. We do not document this type of JVM exception anywhere else IIRC. If we were to doc this, then it would make sense to doc all callers as well, then this would touch almost everything which would not help anyone IMO.

Reply:
As yours, this is why I want to add throws NegativeArraySizeException tag to this method,  In fact, in order to help  users to use and understand these methods,  I think we should add it to the doc of  all methods like it.